src/builtins/hmr-runtime.js

```diff
@@ -14,7 +14,8 @@ function Module(moduleName) {
 
 module.bundle.Module = Module;
 
-if (!module.bundle.parent && typeof WebSocket !== 'undefined') {
+if (!global.hmrReady && typeof WebSocket !== 'undefined') {
+  global.hmrReady = true;
```

Comment:
Why is this global variable necessary?

Reply:
Because on electron there is a native `require`, so `module.bundle.parent` will never be `undefined` as happens on the browser. I understand that this conditional is here to make sure the HMR is only connected once, so I changed to a global flag, that works on both browser and electron.

Reply:
ah i see. maybe we can check if there is no parent bundle, or the parent bundle isn't a parcel bundle. don't really want to add more global variables.

Reply:
How to check if the parent bundle is not a parcel bundle?

Reply:
I just added an `isParcelRequire` flag to the parcel require function

Reply:
:+1: